src/Microsoft.MobileBlazorBindings.Core/NativeComponentRenderer.cs

```diff
@@ -130,5 +135,111 @@ internal NativeComponentAdapter CreateAdapterForChildComponent(IElementHandler p
             _componentIdToAdapter[componentId] = result;
             return result;
         }
+
+        public static void SetNavigationParameters(IComponent component, Dictionary<string, string> parameters)
+        {
+            if (component == null)
+            {
+                throw new ArgumentNullException(nameof(component));
+            }
+            if (parameters == null || parameters.Count == 0)
+            {
+                //parameters will often be null. e.g. if you navigate with no parameters or when creating a root component.
+                return;
+            }
+
+            foreach (var parameter in parameters)
+            {
+                var prop = component.GetType().GetProperty(parameter.Key);
+
+                if (prop != null)
+                {
+                    var parameterAttribute = prop.GetCustomAttribute(typeof(ParameterAttribute));
+                    if (parameterAttribute == null)
+                    {
+                        throw new InvalidOperationException($"Object of type '{component.GetType()}' has a property matching the name '{parameter.Key}', but it does not have [ParameterAttribute] or [CascadingParameterAttribute] applied.");
+                    }
+
+                    if (TryParse(prop.PropertyType, parameter.Value, out object result))
+                    {
+                        prop.SetValue(component, result);
+                    }
+                    else
+                    {
+                        throw new InvalidCastException($"Unable to set property {parameter.Key} on object of type '{component.GetType()}' The error was: Specified cast is not valid. ");
```

Comment:
I think this should be `InvalidOperationException`. `InvalidCastException` is really for the runtime to throw and not meant for libraries. You can find some discussion about this here: https://stackoverflow.com/a/22453825/31668

Reply:
Fixed, thank you.